src/main/scala/edu/berkeley/ce/rockslicing/SeedJointSelector.scala

```diff
@@ -0,0 +1,243 @@
+                     tolerance: Double = 0.01, stepSize: Double = 0.01): Option[Seq[Joint]] = {
+    if (jointSet.length < numProcessors - 1) {
+      println(s"Error: Not enough joints to generate required number of seed joints. ${jointSet.length} joints but"+
+      s" $numProcessors processors. Try using less processors.")
+      return None
+    }
+    val volumePerProc = rockVolume.volume / numProcessors
+    val seedJoints = cycleJointSet(jointSet, rockVolume, volumePerProc, totalVolume,
+      Seq.empty[Joint], tolerance, numProcessors)
+
+    if (stepSize < 0.01) {
+      println("Error: Unable to find satisfactory seed joints using current joint set. Please try another " +
+        "persistent joint set.")
+      None
+    } else if (seedJoints.length != numProcessors - 1) {
+      val newStepSize = if (tolerance >= 1.0) { stepSize / 2.0 } else { stepSize }
+      val newTolerance = if (tolerance >= 1.0) { newStepSize } else { tolerance + stepSize }
+      println(s"Warning: Could not find the required ${numProcessors - 1}, could only find ${seedJoints.length}")
+      println(s"Re-running with lower tolerance: %.1f%%".format(newTolerance * 100.0))
+      findSeedJoints(jointSet, rockVolume, numProcessors, totalVolume, newTolerance, newStepSize)
+    } else {
+      val initialBlock = Seq(rockVolume)
+      val blocks = seedJoints.foldLeft(initialBlock) { (currentBlocks, joint) =>
+        currentBlocks.flatMap(_.cut(joint))
+      }
+      val nonRedundantBlocks = blocks map { case block @ Block(blockCenter, _, _) =>
+          Block(blockCenter, block.nonRedundantFaces)
+      }
+      val volumes = nonRedundantBlocks.map(_.volume)
+      val avgVolume = volumes.sum / volumes.length
+      val maxImbalance = if (math.abs(volumes.max - avgVolume) > math.abs(volumes.min - avgVolume)) {
+        math.abs(volumes.max - avgVolume) / avgVolume
+      } else {
+        math.abs(volumes.min - avgVolume) / avgVolume
+      }
+      println("\nLOAD BALANCE INFORMATION:")
+      println(s"Average Volume: %.2f".format(avgVolume))
+      println(s"Max Volume: %.2f".format(volumes.max))
+      println(s"Min Volume: %.2f".format(volumes.min))
+      Some(seedJoints)
+    }
+  }
+
+  /**
+    * Cycles through input joint set to find seed joints.
+    *
+    * @param jointSet Seq of Joints representing a persistent joint set
+    * @param initialBlock Block that should be subdivided by seed joints into approximately equal volumes
+    * @param volumePerProcessor Ideal volume per processor for load balance
+    * @param totalVolume Total volume of rock volume in analysis
+    * @param selectedJoints Seq of Joints that have been identified as seed joints
+    * @param tolerance Acceptable deviation from ideal volume per processor
+    * @param numProcessors Number of processors that will be used in the analysis
+    * @return
+    */
+  @tailrec
+  private def cycleJointSet(jointSet: Seq[Joint], initialBlock: Block, volumePerProcessor: Double,
```

Comment:
The use of `cycle` is confusing. That will probably make people think either of cycles in graphs or cycling as in shifting the elements of a sequence and wrapping around.

What does this function do and how can we name it accordingly?